src/doc/src/reference/specifying-dependencies.md

```diff
@@ -2,85 +2,106 @@
 
 Your crates can depend on other libraries from [crates.io] or other
 registries, `git` repositories, or subdirectories on your local file system.
-You can also temporarily override the location of a dependency — for example,
-to be able to test out a bug fix in the dependency that you are working on
-locally. You can have different dependencies for different platforms, and
-dependencies that are only used during development. Let's take a look at how
-to do each of these.
+
+It's possible to temporarily change the version or path of a dependency, this
+can be useful for checking the new features or bug fixes after some major
+release.
+
+You can specify different dependencies for:
+- Development and release profiles.
+- Different platforms, architectures and operating systems.
+- Crate "features" that can be enabled and disabled.
 
 ### Specifying dependencies from crates.io
 
-Cargo is configured to look for dependencies on [crates.io] by default. Only
-the name and a version string are required in this case. In [the cargo
-guide](../guide/index.md), we specified a dependency on the `time` crate:
+Here is an example of the most common method for specifying dependencies:
 
 ```toml
 [dependencies]
 time = "0.1.12"
 ```
 
-The string `"0.1.12"` is a [semver] version requirement. Since this
-string does not have any operators in it, it is interpreted the same way as
-if we had specified `"^0.1.12"`, which is called a caret requirement.
+Cargo looks for dependencies on [crates.io] by default. For this case, it only
+requires the name and a version string.
+
+The strings `"0.1.12"` is a [semver] version requirement, and since it does not
+contain any extra operators, it is interpreted as the caret requirement
+`"^0.1.12"`, so that line will download the crate `time` with a version that is
+compatible with `0.1.12` following the Caret requirement rules.
+
+There are five types of version requirements:
+
+Type | Operators | Examples
+-------|-------|-------
+Caret | None or `^` | `"1.2.3"` and `"^1.2.3"`
+Tilde | `~` | `"~1.2.3"`
+Wildcard | `*` | `"1.2.*"`
+Comparison | `>`, `<` and `>=` | `">1.2.3"`, `"<1.2.3"` and `">=1.2.3"`
+Strict | `=` | `"= 1.2.3"`
 
-[semver]: https://github.com/steveklabnik/semver#requirements
+[semver]: https://github.com/dtolnay/semver
```

Comment:
Changed it to https://doc.rust-lang.org/cargo/reference/resolver.html#semver-compatibility, because it already points to https://semver.org/ in the first line.

So it's like we're pointing to both.